gsinghjay/Third_Person_Bear_RPG
# Basic Dialog System Implementation

## Files Added
1. `Assets/Scripts/Dialog/DialogManager.cs`
2. `Assets/Scripts/Dialog/DialogTrigger.cs`

## Implementation Details

### DialogManager.cs
- Singleton pattern for global access
- Text typing effect system
- Dialog progression handling
- Panel visibility management
- Input handling for dialog progression

### DialogTrigger.cs
- Basic trigger system for dialog initiation
- Test dialog lines
- Player detection using trigger collider

## Unity Editor Steps Completed
1. Created Dialog UI:
   - Used existing Canvas
   - Added DialogPanel (UI Panel)
   - Set Panel color to black with 0.8 alpha
   - Added TextMeshPro Text component
   - Configured text settings (size: 36, color: white, alignment: center)

2. Setup DialogManager:
   - Created DialogManager GameObject
   - Attached DialogManager.cs script
   - Assigned DialogPanel reference
   - Assigned DialogText (TMP) reference

4. Created Test Trigger:
   - Added mesh collider to test object
   - Set collider to "Is Trigger"
   - Attached DialogTrigger script
   - Set to Layer 11 (Interactable)

## Testing Instructions
1. Enter play mode
2. Approach the test trigger object
3. Dialog should appear with typing effect
5. Press SPACE to:
   - Skip current typing animation
   - Progress to next line
6. Dialog should close after last line

## Dependencies
- TextMeshPro package
- Requires Canvas setup
- Requires proper layer setup (Layer 11 for Interactable)
